src/notation/internal/notationinteraction.cpp

```diff
@@ -3597,8 +3600,11 @@ void NotationInteraction::pasteSelection(const Fraction& scale)
         #else
         QClipboard::Mode mode = QClipboard::Selection;
         #endif
-        QString txt = QGuiApplication::clipboard()->text(mode);
-        toTextBase(m_editData.element)->paste(m_editData, txt);
+        const QMimeData* mimeData = QApplication::clipboard()->mimeData();
+        if (mimeData->hasFormat(TextEditData::mimeRichTextFormat)) {
```

Comment:
This seems to break the ability to paste (plain) text that was copied outside MuseScore, so you'd need to add some else-if or so to handle that.

For the rest this looks really good!

Reply:
Oops, thanks for pointing it out. Yes, it did break. I added an else section so that it will paste texts copied outside Musescore. Those texts are stored in `plain/text`, so `QMimeData().text()` should work. 

I also add `mode` back or else Musescore in linux will break (i guess i don't have a linux on my side)